src/components/Filters/index.js

```diff
@@ -0,0 +1,337 @@
+
+    if (!rule) {
+      rule = _.cloneDeep(this.state.edit)
+    }
+    const baseRule = inputModels[rule.name]
+    const newRule = _.assign(_.cloneDeep(baseRule), rule)
+
+    if (newRule.id) {
+      this.editFilterRule(newRule)
+    } else {
+      newRule.id = utils.randomString()
+      this.addFilterRule(newRule)
+    }
+    this.setState({
+      showModal: false,
+      edit: this.generateFreshEdit()
+    })
+  }
+
+  updateSimpleSearch (val) {
+    this.setState({ searchString: val })
+    const { rules } = this.props
+    const existingRule = _.find(rules, { name: sieve.SIMPLE_SEARCH_NAME })
+    if (existingRule) {
+      existingRule.value = val
+      this.editFilterRule(existingRule)
+    } else {
+      this.addFilterRule({
+        name: sieve.SIMPLE_SEARCH_NAME,
+        value: val,
+        id: utils.randomString()
+      })
+    }
+  }
+
+  toggleModal () {
+    this.setState({ showModal: !this.state.showModal })
+  }
+
+  showEditModal (rule) {
+    this.setState({
+      showModal: true,
+      edit: rule
+    })
+  }
+
+  removeRule (rule) {
+    if (rule.name === sieve.SIMPLE_SEARCH_NAME) {
+      this.setState({ searchString: '' })
+    }
+
+    const { rules } = this.props
+
+    const updatedRules = rules.filter(r => r.id !== rule.id)
+    this.props.setRules(updatedRules)
+  }
+
+  handleEditChange (e, attribute) {
+    const update = this.state.edit
+    const value = e.target.value
+    const inputModels = sieve.makeFilterInputs(this.props.schema)
+
+    if (attribute === 'name' && update.name !== value) {
+      update.name = e.target.value
+      update.operator = inputModels[value].availableOperators[0]
+      update.value = ''
+    } else {
+      update[attribute] = e.target.value
+    }
+
+    this.setState({ edit: update })
+  }
+
+  saveView (name, scope) {
+    const { rules } = this.props
+    let { views } = this.props
+
+    const newView = {
+      name,
+      rules,
+      id: utils.randomString(),
+      scope
+    }
+
+    if (!views) {
+      views = {}
+    }
+
+    if (scope === 'global') {
+      if (!views.global) {
+        views.global = []
+      }
+      views[scope].push(newView)
+    } else {
+      if (!views[scope]) {
+        views[scope] = {}
+      }
+      if (!views[scope]) {
+        views[scope] = []
+      }
+      views[scope].push(newView)
+    }
+
+    this.props.setViews(views)
+  }
+
+  deleteView (view) {
+    const { views } = this.props
+
+    if (view.scope === 'global') {
+      views.global = views.global.filter(item => item.id !== view.id)
+    } else {
+      views[view.scope] = views[view.scope].filter(item => item.id !== view.id)
+    }
+
+    this.props.setViews(views)
+  }
+
+  render () {
+    const inputModels = sieve.makeFilterInputs(this.props.schema)
+    const rules = this.props.rules || []
+
+    console.log(inputModels)
+    console.log(this.props)
+
+    return (
+      <div style={{ position: 'relative', marginBottom: 20 }}>
```

Comment:
Any reason to use `style=` here?

Reply:
No good reason! I'll extend Box here instead I think